dev/WindowsAppRuntime_BootstrapDLL/WindowsAppRuntime.Bootstrap.rc

```diff
@@ -0,0 +1,33 @@
+#include <winresrc.h>
+
+#define STR1(x) #x
+#define DELIM STR1(.)
+#define STR2(a,b)     STR1(a) DELIM STR1(b)
+
+#define WAS_RELEASE_MAJOR           1
+#define WAS_RELEASE_MINOR           2
```

Comment:
Don't embed these in the .rc

1. Create `/dev/common/WindowsAppSDK-ProductInfo.h` containing constants

```
// Copyright (c) Microsoft Corporation and Contributors.
// Licensed under the MIT License.

#ifndef __WINDOWSAPPSDK_PRODUCTINFO_H__
#define __WINDOWSAPPSDK_PRODUCTINFO_H__

#ifndef STR2
#define STR1(x) #x
#define VERSION_DELIMIMITER STR1(.)
#define STR2(a,b)     STR1(a) VERSION_DELIMIMITER STR1(b)
#endif

#define WINDOWSAPPSDK_RELEASE_MAJOR                         1
#define WINDOWSAPPSDK_RELEASE_MINOR                         2

#define WINDOWSAPPSDK_PRODUCT_VERSION    WINDOWSAPPSDK_RELEASE_MAJOR, WINDOWSAPPSDK_RELEASE_MINOR, 0, 0
#define WINDOWSAPPSDK_PRODUCT_VERSION_STRING   STR2(WINDOWSAPPSDK_RELEASE_MAJOR, WINDOWSAPPSDK_RELEASE_MINOR)
#define WINDOWSAPPSDK_COMPANY_NAME    "Microsoft Corporation"
#define WINDOWSAPPSDK_LEGAL_COPYRIGHT    "Copyright (c) Microsoft Corporation. All rights reserved."
#define WINDOWSAPPSDK_PRODUCT_NAME    "Windows App SDK"

#endif // __WINDOWSAPPSDK_PRODUCTINFO_H__
```

2. In here line 1.5 `#include <WindowsAppSDK-ProductInfo.h"

That should work as WindowsAppRuntime_BootstrapDLL.vcxproj already has `<AdditionalIncludeDirectories>...$(RepoRoot)\dev\common...`

Reply:
Then change WindowsAppRuntime.Bootstrap.rc to

```
// Copyright (c) Microsoft Corporation and Contributors.
// Licensed under the MIT License.

#include <winresrc.h>
#include <WindowsAppSDK-ProductInfo.h>

#define VERSIONINFO_FILENAME "Microsoft.WindowsAppRuntime.Bootstrap.dll"

VS_VERSION_INFO VERSIONINFO
 PRODUCTVERSION WINDOWSAPPSDK_PRODUCT_VERSION
 FILEFLAGSMASK 0x3fL
 FILEFLAGS 0
BEGIN
    BLOCK "StringFileInfo"
    BEGIN
        BLOCK "040904b0"
        BEGIN
            VALUE "CompanyName", WINDOWSAPPSDK_COMPANY_NAME
            VALUE "FileDescription", VERSIONINFO_FILENAME
            VALUE "InternalName", VERSIONINFO_FILENAME
            VALUE "LegalCopyright", WINDOWSAPPSDK_LEGAL_COPYRIGHT
            VALUE "OriginalFilename", VERSIONINFO_FILENAME
            VALUE "ProductName", WINDOWSAPPSDK_PRODUCT_NAME
            VALUE "ProductVersion", WINDOWSAPPSDK_PRODUCT_VERSION_STRING "\0"
        END
    END
END
```

Reply:
Then you also have a template for everything else in the repo

1. If .rc doesn't exist add a copy of this; If .rc already exists add the #include WindowsAppSDK-ProductInfo.h on down
2. Change the VERSIONINFO_FILENAME value

And then in future releases you only need to change the major+minor constants in the .h (2 values in 1 file) and magic happens

Reply:
Thanks!  Updated